app/state/recurringReservations.js

```diff
@@ -89,6 +92,13 @@ const recurringReservationsReducer = handleActions({
   [actions.changeNumberOfOccurrences]: (state, action) => adjustState({
     ...state, numberOfOccurrences: parseInt(action.payload, 10),
   }),
+  [actionTypes.UI.CLOSE_MODAL]: (state, action) => {
+    const modalType = action.payload;
+    if (modalType === modalTypes.RESERVATION_SUCCESS) {
+      return initialState;
+    }
+    return state;
+  },
```

Comment:
Yeah, I was also thinking about whether it should or not. But maybe it makes more sense to reset the state also after the confirm modal is closed. I made the fix and also rebased already.